### PR TITLE
Add BinaryExistsCheck, Updating behaviour of health to perform as per documentation

### PR DESCRIPTION
**Pull request by wilr**
Details on individual commits.

### CODE REVIEW

**Comment by dhensby** [comment]
Needs rebasing

**Comment by wilr** [comment]
@dhensby have rebased. Looks like several of the fixes happened in parallel to this being here.

File: code/checks/BinaryExistsCheck.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	function __construct($binary = "Member") {
+		$this->binary = $binary;
+	}
+
+	function check() {
+		$command = (PHP_OS == 'WINNT') ? 'where' : 'which';
+		$binary = $this->binary;
+
+		$process = proc_open(
+			"$command $binary",
```
Comment (tractorcow) [inline code/checks/BinaryExistsCheck.php]: Can we use http://php.net/manual/en/function.escapeshellarg.php please?

**Comment by tractorcow** [comment]
This will need another rebase sorry. :)

**Comment by wilr** [comment]
Bah. Robots.

**Comment by wilr** [comment]
Closing this, the PSR2 stuff is a dog to merge

**Pull request closed without merging**